### ISSUE TITLE
Add support for react-typescript-app via template (with DRAFT PROPOSAL)

### ISSUE THREAD

**Issue by sahilrajput03** [ameerthehacker/blazepack#11]
## Situation
I tried something to align my `node_modules` for the support of ts packages like @types/react, @types/react-dom and other favourable such type packages to be installed via package manager yarn using a script(named - `by` for `BlazepackYarn`). Example usage of by program is: `./by add @types/lodash` to add types for lodash for auto completion in IDE.

All features of yarn can be used for e.g.,

```bash
yarn add @types/somePackage
yarn remove @types/somePackage
```

and others as arguments are directly passed to yarn cli, i.e., 

```bash
./by add @types/somePackage
./by remove @types/somePackage
```

etc.

# Refer my solution

Guide @ https://github.com/sahilrajput03/react-typescript-blazepack .

## tl;dr

```bash
git clone https://github.com/sahilrajput03/react-typescript-blazepack react-typescript-app && cd react-typescript-app && rm -rf .git
./by # Installs only @types/react and @types/react-dom required for vscode intellisense and autocompletion
code . # Opens up our project folder in vscode
blazepack # Run blazepack
```

You are ready with typescript-react project running in IDE with all auto-completion features!!

**Comment by ameerthehacker** [comment]
@sahilrajput03 I love this idea but is there a way by which we can avoid node_modules installation and still get auto completion?

**Comment by sahilrajput03** [comment]
> @sahilrajput03 I love this idea but is there a way by which we can avoid node_modules installation and still get auto completion?

I guess loading from globally installed module types would be a better idea, but I am not sure if that would work!!..I will try this in a while and will post the results I get !.
😇

**Comment by sahilrajput03** [comment]
@ameerthehacker atleast vscode doesn't respect globally installed packages/modules by default, I am looking for a feature by which it respects globally installed modules!

![image](https://user-images.githubusercontent.com/31458531/108481465-3eecef80-72be-11eb-9563-7c9bfcfacd08.png)

## Another proof vsocde doesn't respect NODE_PATH environment variable to look for node_modules :(
**Notice:  `node testing.js` works fine even without installing react locally :) **

Is there anyway to tell vscode to respect NODE_PATH environemnt path varialle to look for node_modules just as node itself does (in docs: https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders)

**Comment by ameerthehacker** [comment]
@sahilrajput03 you can give [Triple-Slash Directives](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html) a try. Check if this can accept a url like unpkg to read the types.

**Comment by sahilrajput03** [comment]
It doesn't work from global installed modules way...
![image](https://user-images.githubusercontent.com/31458531/108521132-6a3d0200-72f1-11eb-9b28-98b22c0e52e5.png)

, but I am looking to specify the url way though...

**Comment by ameerthehacker** [comment]
@sahilrajput03 thanks a lot for your efforts, will probably go with installing @types while doing blazepack install
